internal/service/iam/server_certificate.go

```diff
@@ -123,64 +107,31 @@ func resourceServerCertificateCreate(d *schema.ResourceData, meta interface{}) e
 		ServerCertificateName: aws.String(sslCertName),
 	}
 
-	if len(tags) > 0 {
-		createOpts.Tags = Tags(tags.IgnoreAWS())
-	}
-
 	if v, ok := d.GetOk("certificate_chain"); ok {
 		createOpts.CertificateChain = aws.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("path"); ok {
-		createOpts.Path = aws.String(v.(string))
-	}
-
 	log.Printf("[DEBUG] Creating IAM Server Certificate with opts: %s", createOpts)
 	resp, err := conn.UploadServerCertificate(createOpts)
 
-	// Some partitions (i.e., ISO) may not support tag-on-create
-	if createOpts.Tags != nil && verify.CheckISOErrorTagsUnsupported(err) {
-		log.Printf("[WARN] failed creating IAM Server Certificate (%s) with tags: %s. Trying create without tags.", sslCertName, err)
-		createOpts.Tags = nil
-
-		resp, err = conn.UploadServerCertificate(createOpts)
-	}
-
 	if err != nil {
 		return fmt.Errorf("error uploading server certificate: %w", err)
 	}
 
 	d.SetId(aws.StringValue(resp.ServerCertificateMetadata.ServerCertificateId))
 	d.Set("name", sslCertName)
 
-	// Some partitions (i.e., ISO) may not support tag-on-create, attempt tag after create
-	if createOpts.Tags == nil && len(tags) > 0 {
-		err := serverCertificateUpdateTags(conn, d.Get("name").(string), nil, tags)
-
-		// If default tags only, log and continue. Otherwise, error.
-		if v, ok := d.GetOk("tags"); (!ok || len(v.(map[string]interface{})) == 0) && verify.CheckISOErrorTagsUnsupported(err) {
-			log.Printf("[WARN] failed adding tags after create for IAM Server Certificate (%s): %s", d.Id(), err)
-			return resourceServerCertificateRead(d, meta)
-		}
-
-		if err != nil {
-			return fmt.Errorf("failed adding tags after create for IAM Server Certificate (%s): %w", d.Id(), err)
-		}
-	}
-
 	return resourceServerCertificateRead(d, meta)
 }
 
 func resourceServerCertificateRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*conns.AWSClient).IAMConn
-	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
-	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
 
 	resp, err := conn.GetServerCertificate(&iam.GetServerCertificateInput{
 		ServerCertificateName: aws.String(d.Get("name").(string)),
 	})
 
-	if tfawserr.ErrCodeEquals(err, iam.ErrCodeNoSuchEntityException) {
+	if tfawserr.ErrCodeEquals(err, iam.ErrCodeNoSuchEntityException) || tfawserr.ErrCodeEquals(err, iam.ErrCodeServerCertificateNotFoundException) {
```

Comment:
```suggestion
	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, iam.ErrCodeServerCertificateNotFoundException) {
```

Такая проверка допустима только для уже созданных сертификатов. На создании пусть падает )

Также можно оставить проверку только кода, который возвращает облако

Reply:
Понял, поправил